### PR TITLE
dynamodb table resource

### PR DESCRIPTION
**Pull request by samscottchef**
Signed-off-by: Sam Scott <sscott@chef.io>

### Description

Building out more resources as per this list here https://github.com/inspec/inspec-aws/issues/145

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_dynamodb_table.rb [hunk 1]
```diff
@@ -12,63 +12,33 @@ class AwsDynamoDbTable < AwsResourceBase
...
-        gsi = { global_secondary_indexes: @dynamodb_table[:global_secondary_indexes] }
-        create_resource_methods(gsi)
-        global_secondary_indexes.each do |idx|
-          idx.define_singleton_method(:to_s) do
-            "AWS Dynamodb Global Secondary Index #{idx.item[:index_name]}"
-          end
-          idx.define_singleton_method(:key_schema) do
-            idx.item[:key_schema].map { |attr| { 'attribute_name': attr[:attribute_name], 'key_type': attr[:key_type] } }
-          end
-        end
```
Comment (rmoles) [inline libraries/aws_dynamodb_table.rb:62]: Is all of this information still available in the response when `create_resource_methods` is used?

Comment (rmoles) [reply libraries/aws_dynamodb_table.rb:62]: If we are removing some of these properties or if they are accessible in a different way, we should probably be updating the docs: https://github.com/inspec/inspec-aws/blob/526369afeebff32aac10315c10090d9f91f2b7ba/docs/resources/aws_dynamodb_table.md#properties

Comment (samscottchef) [reply libraries/aws_dynamodb_table.rb]: Ended up reverting to how it was originally done and just extended it for the sse_description to figure out if it is encrypted.

File: libraries/aws_dynamodb_table.rb [hunk 2]
```diff
@@ -49,7 +51,7 @@ def initialize(opts = {})
                            'key_type':       key.key_type })
       }
 
-      if @dynamodb_table[:global_secondary_indexes]
+      if c[:global_secondary_indexes]
```
Comment (rmoles) [inline libraries/aws_dynamodb_table.rb:54]: is this change intentional?

File: libraries/aws_dynamodb_table.rb [hunk 3]
```diff
@@ -64,6 +66,11 @@ def initialize(opts = {})
     end
   end
 
+  def encrypted?
+    return false unless @dynamodb_table[:sse_description]
+    @dynamodb_table[:sse_description][:status].upcase == 'ENABLED'
```
Comment (rmoles) [inline libraries/aws_dynamodb_table.rb:71]: This could probably be 
```
@dynamodb_table.dig(:sse_description, :status).upcase == 'ENABLED' || false
```

Comment (samscottchef) [reply libraries/aws_dynamodb_table.rb:71]: Looks good however a user could leave out the sse_description completely as its optional, the dig would return nil and calling upcase would get an undefined method

File: libraries/aws_dynamodb_table.rb [hunk 4]
```diff
@@ -64,6 +66,10 @@ def initialize(opts = {})
     end
   end
 
+  def encrypted?
```
Comment (rmoles) [inline libraries/aws_dynamodb_table.rb:69]: Can we add some documentation for this?

Comment (samscottchef) [reply libraries/aws_dynamodb_table.rb:69]: updated singular doc

**Review by rmoles — APPROVED** [review]
Good work. Thanks @samscottchef

**Review by skpaterson — APPROVED** [review]
Thanks @samscottchef !
